internal/pkg/addon/addons.go

```diff
@@ -31,8 +31,8 @@ var (
 )
 
 type workspaceReader interface {
-	ReadAddonsDir(svcName string) ([]string, error)
-	ReadAddon(svcName, fileName string) ([]byte, error)
+	ReadWorkloadAddonsDir(svcName string) ([]string, error)
+	ReadWorkloadAddon(svcName, fileName string) ([]byte, error)
```

Comment:
Instead of `ReadWorkloadAddonsDir` vs `ReadEnvAddonsDir`, what do you think of the following API?

```go
package workspace

// ListAddonsFiles returns a list of file paths to all the addons templates defined under the dir.
func (ws *Workspace) ListAddonsFiles(dirPath string) ([]string, error)

// ReadAddonsFile reads the addons template file under fpath.
func (ws *Workspace) ReadAddonsFile(fpath string) ([]byte, error)

// EnvAddonsPath returns the addons/ directory file path for environments.
func (ws *Workspace) EnvAddonsPath() string

// WorkloadAddonsPath returns the addons/ directory file path for a given workload.
func (ws *Workspace) WorkloadAddonsPath(name string) string
```
The idea is to have a generic `ListAddonsFiles` that can be given as input the `EnvAddonsPath` or `WorkloadAddonsPath`.

Reply:
Addressed! 

I actually had:
```
WorkloadAddonsPath(name string) (string, error)
WorkloadAddonFilePath(wkldName, fName string) (string, error)
EnvAddonsPath() string
EnvAddonFilePath() string
ListFiles(dirPath string) ([]string, error)
ReadFile(fPath string) ([]byte, error)
```
So that we are still letting the `workspace` package to determine how an addon file is arranged under an `addons/` directory. lmk how you like it!